cli/src/lib/context.ts

```diff
@@ -0,0 +1,64 @@
+import os from "os";
+import path from "path";
+import fs from "fs";
+
+export type CliContext = {
+  apiUrl?: string;
+  consoleUrl?: string;
+  cluster?: string;
+  service?: string;
+  deployment?: string;
+};
+
+const BASE_CONTEXT_PATH = path.join(os.homedir(), ".differential");
```

Comment:
Nice one!

Another idea:

Another options is to save context per Differential project and instead save the `.differential` file in the directory where the user is doing the operation?

So, when a command is run, it gets the context from the directory from where it's being run from.

It could even be a `differential.toml` and we can follow what fly does.

Reply:
Sounds good. Will move this to per-project for now and then look at toml, etc later.

Reply:
I have updated this to move the context files into the project directory.
Also, the named contexts now override the exiting ones. `defaults` -> `differential.json` -> `differential.local.json` etc.

Json for now but will look at replacing with yaml / toml.